### DISCUSSION TITLE
Switch to front camera.?

### DISCUSSION THREAD

**Discussion by Arun1947** [sceneview/sceneform-android#213] [Q&A] [answered]
Is there any method to flip the Scene Form camera to primary and secondary (Front & Back)?

**Comment by RGregat** [comment]
You could take a look into the AugmentedFaces implemenation. It changes the used camera to **FRONT**

https://github.com/SceneView/sceneform-android/blob/5006b55f0fd3d983eebc52d916d5ea166a1ded0a/ux/src/main/java/com/google/ar/sceneform/ux/ArFrontFacingFragment.java#L24

**Comment by Arun1947** [comment]
Hi, @RGregat 
This code worked and succesfully switched to front_camera, but can't go back to back_camera while applying
filter.setFacingDirection(CameraConfig.FacingDirection.BACK); 

gives an exception -- com.google.ar.core.exceptions.UnavailableException sceneform (Failed to create AR session)

**Comment by ThomasGorisse** [comment]
You canot change the Camera settings while the Session is active.
You will have to recreate it in order to do the change.

**Comment by Arun1947** [comment]
@ThomasGorisse No, the session was not active, I added camera as Back_camera at the initial time.

![image](https://user-images.githubusercontent.com/94825638/143032331-edf5ef6e-6b75-4d3f-b5a4-8f52f88e0945.png)

**Comment by ThomasGorisse** [comment]
@fvito @grassydragon @doranteseduardo Does the AugmentedFace could work on the back Camera?

**Comment by doranteseduardo** [comment]
No, according to the ARCore docs:


"Augmented Faces supports front-facing (selfie) camera only...".

**Comment by Arun1947** [comment]
@doranteseduardo @ThomasGorisse Does the Augmented-features (other than AugmentedFace) could work on the front Camera?

**Comment by ThomasGorisse** [comment]
Never tried it but you can have a test and tell us here.

**Comment by doranteseduardo** [comment, marked as answer]
> @doranteseduardo @ThomasGorisse Does the Augmented-features (other than AugmentedFace) could work on the front Camera?

No, when the selfie camera is used only augmented faces feature work, also, the image gets mirrored, hitTest returns an empty array, no planes are detected and so-on.

When using the rear camera, all other features work except for the augmented faces.

**Comment by sarah06jacobs** [comment]
Can we get some code for switching cameras?  I've been at this for days.  I tried pausing the session:
```

    CameraConfigFilter filter = new CameraConfigFilter(session);
            filter.setFacingDirection(CameraConfig.FacingDirection.FRONT);

            session.pause();
            new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
                @Override
                public void run() {
                    session.setCameraConfig(session.getSupportedCameraConfigs(filter).get(0));
                    try {
                        session.resume();
                    } catch (CameraNotAvailableException ex) {
                        session = null;
                        return;
                    }
                }
```

but this crashes, because session.update gets called after session.pause.

I also tried removing and re-adding the fragment, but when I go back to back facing camera, the camera is blurry.
```

      removeARFragment();
        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
            @Override
            public void run() {
                addARFragment();
            }
        }, 100);

       public void removeARFragment() {
        FragmentManager fm = getSupportFragmentManager();
        ArFragment arFragment = (ArFragment) fm.findFragmentByTag(TAG_AR_FRAGMENT);
        if (arFragment != null) {
            arFragment.destroySession();
            fm.beginTransaction().remove(arFragment).commit();
        }
    }

@Override
    public void addARFragment() {
        FragmentManager fm = getSupportFragmentManager();
        this.arFragment = (ArFragment) fm.findFragmentByTag(TAG_AR_FRAGMENT);
        if (this.arFragment == null) {
            this.arFragment = ArFragment.newInstance(true);
            fm.beginTransaction().replace(R.id.contentARContainer, this.arFragment, TAG_AR_FRAGMENT).commit();
        }
    }

```
